code/modules/events/ion_storm.dm

```diff
@@ -11,10 +11,12 @@
 	var/shuffleLawsChance = 10 //chance the AI's laws are shuffled afterwards
 	var/botEmagChance = 1
 	var/ionMessage = null
+	var/lawsource = "Ion Storm"
 	announceWhen	= 1
 	announceChance = 33
 
 /datum/round_event/ion_storm/add_law_only // special subtype that adds a law only
+	lawsource = "unspecified, please report this to coders"

```

Comment:
Why is the source here unspecified?

Reply:
@qwertyquerty because it is expected that whoever will code some other reasons for ion law addition will specify a reason. If he does not, this will say he is a bad boy. /datum/round_event/ion_storm/add_law_only is only called when there is the need for just the law.